eeps/eep-XXX.md

```diff
@@ -0,0 +1,278 @@
+is of a different basic type, for example `atom()`, Dialyzer will reject it.
+
+    -spec bar() -> foot().
+    bar() -> meter_to_foot(24). 
+    % Input type: integer(). 
+    % Expected type: meter().
+    % Result: No error.
+
+When a nominal type is expected, passing in a type with the same structure
+is also allowed.  In the following example, Dialyzer does not return any
+error for the function `qaz()`.  The spec for `meter_ctor()` expects a
+`integer()` type as input. Passing in a `meter()` type is allowed, because
+`meter()` is defined to have type `integer()`.  Similarly, passing in a
+type `foot()` is also allowed when `integer()` is expected.
+
+    -spec qaz() -> integer().
+    qaz() -> meter_ctor(meter_ctor(24)). 
+    % Input of the outer meter_ctor(): meter(). 
+    % Expected type: integer().
+    % Result: No error. 
+
+It is also worth noting that nested nominal types are supported.  In the
+following example:
+
+    -nominal state() :: integer().
+    -nominal container() :: state().
+    -nominal record_container() :: #{a => state(), b => [container()|atom()]}.
+
+Nominal type-checking in Dialyzer correctly recognizes that `container()`
+can be safely used whenever a type `state()` is expected, just as `state()`
+can be safely used when a type `integer()` is expected.  Using `[state()]`
+as the second parameter of `record_container()`'s construction is allowed,
+even though `container()` and `state()` have different names.
+
+Nominal Type-Checking Rules
+----------------------------
+
+In order to specify nominal type-checking rules, there are three terms that
+need to be defined. The scope of these definitions are local to this EEP.
+
+For two nominal types `s()` and `t()`, `s()` is a **nominal subtype** of
+`t()`, and `t()` is a **nominal supertype** of `s()` if `t()` is nested
+in `s()`.
+
+- Cases where `s()` is a nominal subtype of `t()`:
+    - `t()` can be directly nested in `s()`.
+
+          -nominal s() :: t().
+    - `t()` can be nested in other nominal type(s), which is then nested
+    in `s()`.
+
+          -nominal s() :: nominal_1().
+          -nominal nominal_1() :: nominal_2().
+          -nominal nominal_2() :: t().
+- One case where `s()` is *not* a nominal subtype of `t()`:
+    - `s()` is defined as a union type of `t()` and some other type. The
+    nesting cannot contain other types before reaching `t()`.
+
+          -nominal s() :: t() | 'foo'.
+
+A non-nominal type is **compatible** with a nominal or non-nominal type
+if they share common values. Empirically, the function `erl_types:t_inf/2`
+can compute the intersection of 2 types. Two types that have a non-empty
+intersection are structurally compatible.
+
+- Examples:
+    - 4711 and 42 are not structurally compatible.
+    - 4711 and `integer()` are structurally compatible. (Their intersection
+    is the value 4711.)
+    - `list(any_type)` and `[]` are structurally compatible.
```

Comment:
Since you write the reason, may it helps the reader if you write their intersection as in the prev. point above